app/src/main/java/com/sunasterisk/moviedb_51/ui/home/HomeFragment.kt

```diff
@@ -1,8 +1,80 @@
 package com.sunasterisk.moviedb_51.ui.home
 
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.databinding.DataBindingUtil
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import com.sunasterisk.moviedb_51.R
+import com.sunasterisk.moviedb_51.data.base.BaseViewModelFactory
+import com.sunasterisk.moviedb_51.data.repository.MovieRepository
+import com.sunasterisk.moviedb_51.data.source.local.MovieLocalDataSource
+import com.sunasterisk.moviedb_51.data.source.remote.MovieRemoteDataSource
+import com.sunasterisk.moviedb_51.data.source.remote.api.MovieFactory
+import com.sunasterisk.moviedb_51.databinding.FragmentHomeBinding
+import com.sunasterisk.moviedb_51.utils.NetworkUtil
 
 class HomeFragment : Fragment() {
+    private lateinit var viewModel: HomeViewModel
+    private lateinit var binding: FragmentHomeBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val movieRepository =
+            MovieRepository.getInstance(
+                MovieRemoteDataSource.getInstance(MovieFactory.instance),
+                MovieLocalDataSource.instance
+            )
+        viewModel = ViewModelProvider(
```

Comment:
```suggestion
        viewModel = ViewModelProviders(
```

Reply:
em đọc được là ViewModelProviders không được dùng nữa. link:https://developer.android.com/reference/androidx/lifecycle/ViewModelProviders 
"viewmodelproviders' is deprecated. deprecated in java"